src/Moryx/Configuration/ConfigManagerExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Moryx.Configuration
+{
+    public static class ConfigManagerExtensions
+    {
+        /// <summary>
+         /// Get typed configuration. Will use cached object if available
+         /// </summary>
+         /// <typeparam name="T">Type of config object</typeparam>
+         /// <returns>Configuration object</returns>
+        public static T GetConfiguration<T>(this IConfigManager configManager) 
+            where T : class, IConfig, new()
+        {
+            var configType = typeof(T);
+            return (T)configManager.GetConfiguration(configType, configType.FullName, false);
+        }
+
+        /// <summary>
+        /// Get typed configuration. Will use cached object if available
+        /// </summary>
+        /// <typeparam name="T">Type of config object</typeparam>
+        /// <returns>Configuration object</returns>
+        public static T GetConfiguration<T>(this IConfigManager configManager, string name)
+            where T : class, IConfig, new()
+        {
+            var configType = typeof(T);
+            return (T)configManager.GetConfiguration(configType, name, false);
+        }
+
+        /// <summary>
+        /// Get typed configuration. Also specifies behaviour for implementations with internal cache
+        /// </summary>
+        /// <typeparam name="T">Type of config object</typeparam>
+        /// <param name="getCopy"><value>True</value>Create new instance. <value>False</value>Get from cache if possible</param>
+        /// <returns>Configuration object</returns>
+        public static T GetConfiguration<T>(this IConfigManager configManager, bool getCopy)
```

Comment:
Github demands xml tags for the configManager parameter on all the methods. I don't know how this should be done properly, but if its a warning in CS as well, we should fix ist 😊